### PR TITLE
Use ArgumentOutOfRangeException Throw methods in SendFileResponseExtensions..CheckRange

### PR DESCRIPTION
**Pull request by paulomorgado**
To allow inlining by the JIT.

# Use ArgumentOutOfRangeException Throw methods in SendFileResponseExtensions..CheckRange to allow inlining by the JIT

- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [x] You've included unit or integration tests for your change, where applicable.
- [x] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

Methods that throw exceptions are not inlinable by the JIT.

This PR uses the recently introduced **throw** methods in the `ArgumentOutOfRangeException` to allow the logic to be fully inlined in non-error situations to increase performance of static file responses.

### CODE REVIEW

File: src/Http/Http.Extensions/src/SendFileResponseExtensions.cs [hunk 1]
```diff
@@ -124,14 +124,11 @@ private static async Task SendFileAsyncCore(HttpResponse response, string fileNa
...
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(offset, fileLength);
+        if (count is {} countValue)
         {
-            throw new ArgumentOutOfRangeException(nameof(offset), offset, string.Empty);
-        }
-        if (count.HasValue &&
-            (count.GetValueOrDefault() < 0 || count.GetValueOrDefault() > fileLength - offset))
-        {
-            throw new ArgumentOutOfRangeException(nameof(count), count, string.Empty);
+            ArgumentOutOfRangeException.ThrowIfGreaterThan(countValue, countValue, nameof(count));
```
Comment (captainsafia) [inline src/Http/Http.Extensions/src/SendFileResponseExtensions.cs:131]: I might be missing something but it appears we've lost the `count > fileLength - offset` check here?

Comment (paulomorgado) [reply src/Http/Http.Extensions/src/SendFileResponseExtensions.cs]: @captainsafia,

> I might be missing something but it appears we've lost the `count > fileLength - offset` check here?

Nope! My mistake! Fixed!

File: src/Http/Http.Extensions/src/SendFileResponseExtensions.cs [hunk 2]
```diff
@@ -124,14 +124,11 @@ private static async Task SendFileAsyncCore(HttpResponse response, string fileNa
...
+        ArgumentOutOfRangeException.ThrowIfGreaterThan(offset, fileLength);
+        if (count is {} countValue)
         {
-            throw new ArgumentOutOfRangeException(nameof(offset), offset, string.Empty);
-        }
-        if (count.HasValue &&
-            (count.GetValueOrDefault() < 0 || count.GetValueOrDefault() > fileLength - offset))
-        {
-            throw new ArgumentOutOfRangeException(nameof(count), count, string.Empty);
+            ArgumentOutOfRangeException.ThrowIfGreaterThan(countValue, fileLength - offset, nameof(count));
```
Comment (amcasey) [inline src/Http/Http.Extensions/src/SendFileResponseExtensions.cs:131]:
```suggestion
            ArgumentOutOfRangeException.ThrowIfLessThan(countValue, 0, nameof(count));
            ArgumentOutOfRangeException.ThrowIfGreaterThan(countValue, fileLength - offset, nameof(count));
```

Comment (amcasey) [reply src/Http/Http.Extensions/src/SendFileResponseExtensions.cs:131]: Or maybe I'm missing something?

**Review by amcasey — APPROVED** [review]

**Comment by amcasey** [comment]
FYI @captainsafia, I've enable auto-merge, since it seems straightforward.  You have two hours to shout if you disagree. :wink:

**Review by captainsafia — APPROVED** [review]

**Comment by paulomorgado** [comment]
@captainsafia, @amcasey, looks like the build still has issues.

**Comment by amcasey** [comment]
@paulomorgado When that happens, you should be able to unblock yourself by closing and re-opening the PR.  (We're happy to help, but you don't have to wait for us.)  Obviously, we're working on making it happen less often.

**Comment by amcasey** [comment]
But you don't need to do it after we manually restart the tests with `/azp run`. 😛

**Comment by amcasey** [comment]
Ack, apparently close and re-open disables auto-merge.  TIL.  Sorry about that.

**Pull request merged**